### ISSUE TITLE
fmt::format_to_n  hangs when argument length exceeds over buffer with provided size.

### ISSUE THREAD

**Issue by arthurfait** [fmtlib/fmt#2028]
Following code fragment hangs
```c++
#include <fmt/format.h>

int main()
{
    char buff[80];
    const char *a = "111111111111111111111111111111111111111111111111111111111111111111111111111111111";
    fmt::format_to_n(std::begin(buff), 79,FMT_STRING("{} "), a);
    fmt::print("done");
    return 0;
}
```

Seems there is some endless loop.

Most obvious:
```c++
template <typename T>
template <typename U>
void buffer<T>::append(const U* begin, const U* end) {
  do {
    auto count = to_unsigned(end - begin);
    try_reserve(size_ + count); // << mostly nop for fixed buffers
    auto free_cap = capacity_ - size_;
    if (free_cap < count) count = free_cap;
    std::uninitialized_copy_n(begin, count, make_checked(ptr_ + size_, count));
    size_ += count;
    begin += count;
  } while (begin != end);
}
```

**Comment by vitaut** [comment]
Already fixed in master (#1996).

**Comment by arthurfait** [comment]
awaiting next release
